lib/msal-core/src/PopTokenGenerator.ts

```diff
@@ -0,0 +1,153 @@
+import { BrowserCrypto } from './BrowserCrypto';
+import { Base64 } from 'js-base64';
+import { HashAlgorithm, getStringFromArrayBuffer, KeyFormat, utf8Encode, getArrayBufferFromString, KeyUsages, KeyGenAlgorithm, getJwkString } from './PopTokenCommon';
+import { Utils } from './Utils';
+import { DatabaseCache } from './DatabaseCache';
+import { AuthenticationParameters } from './AuthenticationParameters';
+
+type CachedKeyPair = {
+    publicKey: CryptoKey,
+    privateKey: CryptoKey,
+    requestMethod: string,
+    requestUri: string
+}
+
+export class PopTokenGenerator {
```

Comment:
This class so I understand, is essentially a browser based glue class for  Crypto, and Database interfaces? 
if so, should this be interacting with interfaces and constructed with implementations?

Reply:
Yeah I thought about that (especially thinking ahead to Node), but decided not to overcomplicate things to start. Can always do that incrementally.